x-pack/plugins/security_solution/common/api/timeline/model/components.gen.ts

```diff
@@ -0,0 +1,350 @@
+    .optional(),
+  match_all: z.string().optional(),
+  missing: z.string().optional(),
+  query: z.string().optional(),
+  range: z.string().optional(),
+  script: z.string().optional(),
+});
+
+export type SerializedFilterQueryResult = z.infer<typeof SerializedFilterQueryResult>;
+export const SerializedFilterQueryResult = z.object({
+  filterQuery: z
+    .object({
+      kuery: z
+        .object({
+          kind: z.string().optional(),
+          expression: z.string().optional(),
+        })
+        .optional(),
+      serializedQuery: z.string().optional(),
+    })
+    .optional(),
+});
+
+export type Sort = z.infer<typeof Sort>;
+export const Sort = z.object({
+  columnId: z.string().optional(),
+  sortDirection: z.string().optional(),
+});
+
+export type SavedTimeline = z.infer<typeof SavedTimeline>;
+export const SavedTimeline = z.object({
+  columns: ColumnHeaderResult.optional(),
+  created: z.number().optional(),
+  createdBy: z.string().optional(),
+  dataProviders: z.array(z.lazy(() => DataProviderResult)).optional(),
+  dataViewId: z.string().optional(),
+  dateRange: z
+    .object({
+      end: z.union([z.string(), z.number()]).optional(),
+      start: z.union([z.string(), z.number()]).optional(),
+    })
+    .optional(),
+  description: z.string().optional(),
+  eqlOptions: z
+    .object({
+      eventCategoryField: z.string().optional(),
+      tiebreakerField: z.string().optional(),
+      timestampField: z.string().optional(),
+    })
+    .optional(),
+  eventType: z.string().optional(),
+  excludedRowRendererIds: z.array(RowRendererId).optional(),
+  favorite: z.array(FavoriteTimelineResult).optional(),
+  filters: z.array(FilterTimelineResult).optional(),
+  kqlMode: z.string().optional(),
+  kqlQuery: SerializedFilterQueryResult.optional(),
+  indexNames: z.array(z.string()).optional(),
+  savedQueryId: z.string().optional(),
+  sort: Sort.optional(),
+  status: z.enum(['active', 'draft', 'immutable']).optional(),
+  title: z.string().optional(),
+  templateTimelineId: z.string().optional(),
+  templateTimelineVersion: z.number().optional(),
+  timelineType: TimelineType.optional(),
+  updated: z.number().optional(),
+  updatedBy: z.string().optional(),
+});
+
+export type BareNote = z.infer<typeof BareNote>;
+export const BareNote = z.object({
+  eventId: z.string().optional(),
+  note: z.string().optional(),
+  timelineId: z.string().optional(),
+  created: z.number().optional(),
+  createdBy: z.string().optional(),
+  updated: z.number().optional(),
+  updatedBy: z.string().optional(),
+});
+
+export type Note = z.infer<typeof Note>;
+export const Note = BareNote.merge(
+  z.object({
+    noteId: z.string().optional(),
+    version: z.string().optional(),
+  })
+);
+
+export type PinnedEvent = z.infer<typeof PinnedEvent>;
+export const PinnedEvent = z.object({
+  pinnedEventId: z.string().optional(),
+  eventId: z.string().optional(),
+  timelineId: z.string().optional(),
+  created: z.number().optional(),
```

Comment:
A few of these fields are nullable as well

Reply:
[Fixed here](https://github.com/elastic/kibana/pull/188844/commits/f70299c8c1e32d4b0d22ccb3580ae918520db71a)